### PR TITLE
Pinmux mec1501

### PR DESCRIPTION
**Pull request by franciscomunoz**
Notes: The HW contains GPIO settings and Pinmuxing in the same registers for each pin. Each pinmux base address is coming from the existing GPIO base addresses (see the dts_fixup). This change also fixes a formatting issue in a table from the documentation. Lastly, serial console is moved to UART2 in order to use the ARM tracing features.

### CODE REVIEW

File: boards/arm/mec15xxevb_assy6853/doc/index.rst [hunk 1]
```diff
@@ -148,26 +148,28 @@ loads from SHD SPI.
...
 --------------------------
 
-Each column of the following table illustrates how to enable UART0, JTAG,
+Each column of the following table illustrates how to enable SWD,
 PVT SPI, SHD SPI and LED0-2 respectively.
 
 +---------+--------+-----------+----------+---------+
-| JP68    | JP9    | JP38      | JP98     | JP41    |
-| (UART0) | (JTAG) | (PVT SPI) | (SHD SPI)| (LED0-2)|
+| JP9     | JP9    | JP38      | JP98     | JP41    |
```
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Should these both be JP9?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/mec15xxevb_assy6853/doc/index.rst (hunk 1 above)
Comment (franciscomunoz) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: Correct both features are controlled from JP9

File: drivers/pinmux/pinmux_mchp_xec.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	} else {
+		/* None : Pin tristates when no active driver is present
+		 * on the pin. This is the POR setting
+		 */
+		pcr1 |= MCHP_GPIO_CTRL_PUD_NONE;
+	}
+
+	/* Make sure gpio isrs are disabled */
+	pcr1 |= MCHP_GPIO_CTRL_IDET_DISABLE;
+	mask |= MCHP_GPIO_CTRL_IDET_DISABLE;
```
Comment (scottwcpg) [inline drivers/pinmux/pinmux_mchp_xec.c]: MCHP GPIO interrupt detect is a multi-bit field in the GPIO control register.  The mask should be OR's with MCHP_GPIO_CTRL_IDET_MASK.

**Review by dbkinder — APPROVED** [review]
OK then, thanks.

File: drivers/pinmux/pinmux_mchp_xec.c (hunk 2 above)
Comment (franciscomunoz) [reply drivers/pinmux/pinmux_mchp_xec.c]: I changed it.

Comment (scottwcpg) [reply drivers/pinmux/pinmux_mchp_xec.c]: OK. Thanks

**Review by scottwcpg — APPROVED** [review]

**Review by albertofloyd — APPROVED** [review]

**Pull request merged**